File: python/perf/perf.py
import sys
import time
sys.path.append("..")

from motifcluster import clustering as mccl
from motifcluster import motifadjacency as mcmo
from motifcluster import sampling as mcsa
from motifcluster import utils as mcut

import numpy as np

def test_perf():

  for n in [100, 200, 500, 1000, 2000]:
    for p in [10/n, 100/n]:
      for motif_name in ["Ms", "Md", "M1", "M9", "M11"]:

        t0 = time.time()

        block_sizes = [n]
        connection_matrix = np.array([p]).reshape((1, 1))
        weight_matrix = np.array([10]).reshape((1, 1))

        adj_mat = mcsa.sample_dsbm(block_sizes, connection_matrix,
                                   weight_matrix)

        mcmo.build_motif_adjacency_matrix(adj_mat, motif_name)

        #print(n, p, motif_name)
        #print("{:.2f}".format(time.time() - t0), "\n")

  return None

File: R/perf/perf.R
library(motifcluster)

t00 = Sys.time()
for (n in c(100, 200, 500, 1000, 2000)) {
  for (p in c(10/n, 100/n)){
    for (motif_name in c("Ms", "Md", "M1", "M9", "M11")){

      t0 = Sys.time()
      block_sizes = c(n)
      connection_matrix = matrix(p)
      weight_matrix = matrix(10)

      adj_mat = sample_dsbm(block_sizes, connection_matrix,
                            weight_matrix)

      build_motif_adjacency_matrix(adj_mat, motif_name)

      cat(n, p, motif_name, "\n")
      cat(round(Sys.time() - t0, 2), "\n\n")
    }
  }
}

cat("Total time: \n")
cat(round(Sys.time() - t00, 2)"\n")
